### PR TITLE
Addition of conan to conda.

### PR DESCRIPTION
**Pull request by windelbouwmanbosch**
This package still has a lot of dependencies, but these are in seperate pull requests.

One question I have is about the patch module. It is a python package, but there also exists a non-python package. How should this be handled? Should I add python-patch under this name?

### CODE REVIEW

**Comment by windelbouwmanbosch** [comment]
I think this PR is now ready. All dependencies have been solved.

**Comment by windelbouwmanbosch** [comment]
Could this be merged?

**Comment by dirmeier** [comment]
very cool!

**Comment by abloemert** [comment]
@synapticarbors @CJ-Wright can you please merge this pull request?

File: recipes/conan/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+build:
+  number: 0
+  entry_points:
+    - conan=conans.conan:run
+    - conan_server=conans.conan_server:run
+    - conan_build_info=conans.build_info.command:run
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline recipes/conan/meta.yaml]: Are all of these packages required to `build` this recipe? Typically when using `--single-version-externally-managed`, you only need `python` and `setuptools`, unless there is a `setup_requires` section in the `setup.py` or some packages get imported by that same script.

File: recipes/conan/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  home: https://conan.io
+  license: MIT
+  license_family: MIT
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE.md'
+  summary: Conan C/C++ package manager
+  doc_url: http://docs.conan.io/en/latest/
+  dev_url: https://github.com/conan-io/conan
+
+extra:
+  recipe-maintainers:
```
Comment (synapticarbors) [inline recipes/conan/meta.yaml]: Please get permission from the co-maintainers. If you know then personally and have had conversations offline, you can clarify that here, or you need to ping them and get their explicit consent.

File: recipes/conan/meta.yaml (hunk 1 above)
Comment (windelbouwmanbosch) [reply recipes/conan/meta.yaml]: Okay, that makes sense. I did not know this. Thanks!

File: recipes/conan/meta.yaml (hunk 2 above)
Comment (windelbouwmanbosch) [reply recipes/conan/meta.yaml]: Yes, I had a conversation with these people and asked if they would be interested, and they gave their ok.

**Pull request merged**